src/optionsMenu/AdvancedTab.tsx

```diff
@@ -1,14 +1,105 @@
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { RootState } from "../rootReducer";
 import { useSelector, useDispatch } from "react-redux";
 import { changeSetting } from "./settingsState";
 import { changeBroadcastSetting } from "../broadcast/state";
 
+function reduceToOutputs(devices: MediaDeviceInfo[]) {
+  var temp: MediaDeviceInfo[] = [];
+  devices.forEach((device) => {
+    if (device.kind === "audiooutput") {
+      temp.push(device);
+    }
+  });
+  return temp;
+}
+
+function ChannelOutputSelect({
+  outputList,
+  channel,
+}: {
+  outputList: MediaDeviceInfo[] | null;
+  channel: number;
+}) {
+  const settings = useSelector((state: RootState) => state.settings);
```

Comment:
Can we use a more specific selector here? Using just `state.settings` could cause unnecessary re-rendering (not applicable here because `settings` changes relatively rarely, but a good habit to get into ;) )

Reply:
Donee